client/src/components/editapp.js

```diff
@@ -1 +1,136 @@
 
+import React from 'react'
+import axios from 'axios'
+import Input from './input'
+
+class Edit extends React.Component {
+  state = {
+    editForm: false
+  }
+
+  // Update REST route
+  updateApp = (event, formInputs) => {
+    event.preventDefault()
+    const id = event.target.getAttribute('id')
+    axios
+      .put('/applications/' + id, formInputs)
+      .then(
+          response => {
+            this.setState({
+              apps: response.data
+            })
+          }
+      )
+  }
+
+  editStatus = event => {
+    this.setState({
+      updateStatus: event.target.value
+    })
+  }
+
+  editDate = event => {
+    this.setState({
+      updateDate: event.target.value
+    })
+  }
+
+  editTitle = event => {
+    this.setState({
+      updateTitle: event.target.value
+    })
+  }
+
+  editCompany = event => {
+    this.setState({
+      updateCompany: event.target.value
+    })
+  }
+
+  editLocation = event => {
+    this.setState({
+      updateLocation: event.target.value
+    })
+  }
+
+  editLink = event => {
+    this.setState({
+      updateLink: event.target.value
+    })
+  }
+
+  toggleEditForm = () => {
+    this.setState(
+      (prevState) => {
+        return {editForm: !prevState.editForm}
+    })
+  }
+
+
+  render(){
+    const { app } = this.props
+    return(
+      <div>
+        <button onClick={this.toggleEditForm}>
+          Edit
+        </button>
+        { this.state.editForm
+          ?
+          <form id={app._id} onSubmit={this.updateApp}>
+            <Input
```

Comment:
I'll work on refactoring this! But going to leave it like this for now. Might ask for your suggestion on how to go about refactoring this.

Reply:
Sounds good!

Reply:
In the future, rather than merging it on in, I'd recommend to just keep working on the branch, and push to the branch with the changes requested, then I'll be able to see those changes and you can merge it in once you get the thumbs up. 

At the end of the day, this is your repo so you can feel free to do whatever you'd like, but as far as standard workflows, that would be better. Just a tip!